pvlib/iotools/meteonorm.py

```diff
@@ -0,0 +1,723 @@
+        specified, UTC is assumed.
+    api_key : str
+        Meteonorm API key.
+    parameters : list or 'all', default : 'all'
+        List of parameters to request or `'all'` to get all parameters.
+    surface_tilt : float, default : 0
+        Tilt angle from horizontal plane.
+    surface_azimuth : float, default : 180
+        Orientation (azimuth angle) of the (fixed) plane. Clockwise from north
+        (north=0, east=90, south=180, west=270).
+    time_step : {'1min', '15min', '1h'}, default : '15min'
+        Frequency of the time series.
+    horizon : str or list, default : 'auto'
+        Specification of the horizon line. Can be either 'flat', 'auto', or
+        a list of 360 integer horizon elevation angles.
+    interval_index : bool, default : False
+        Index is pd.DatetimeIndex when False, and pd.IntervalIndex when True.
+        This is an experimental feature which may be removed without warning.
+    map_variables : bool, default : True
+        When true, renames columns of the Dataframe to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+    url : str, optional
+        Base URL of the Meteonorm API. The default is
+        :const:`pvlib.iotools.meteonorm.URL`.
+
+    Raises
+    ------
+    requests.HTTPError
+        Raises an error when an incorrect request is made.
+
+    Returns
+    -------
+    data : pd.DataFrame
+        Time series data. The index corresponds to the start (left) of the
+        interval unless ``interval_index`` is set to True.
+    meta : dict
+        Metadata.
+
+    Examples
+    --------
+    >>> # Retrieve historical time series data
+    >>> df, meta = pvlib.iotools.get_meteonorm_observation_training(  # doctest: +SKIP
+    ...     latitude=50, longitude=10,  # doctest: +SKIP
+    ...     start='2023-01-01', end='2025-01-01',  # doctest: +SKIP
+    ...     api_key='redacted',  # doctest: +SKIP
+    ...     endpoint='training')  # doctest: +SKIP
+
+    See Also
+    --------
+    pvlib.iotools.get_meteonorm_observation_realtime
+
+    References
+    ----------
+    .. [1] `Meteonorm
+       <https://meteonorm.com/>`_
+    .. [2] `Meteonorm API
+       <https://docs.meteonorm.com/docs/getting-started>`_
+    .. [3] `Meteonorm API reference
+       <https://docs.meteonorm.com/api>`_
+    """  # noqa: E501
+    endpoint = "observation/training"
+
+    data, meta = _get_meteonorm(
+        latitude,
+        longitude,
+        start,
+        end,
+        api_key,
+        parameters,
+        surface_tilt,
+        surface_azimuth,
+        time_step,
+        horizon,
+        interval_index,
+        map_variables,
+        url,
+        endpoint,
+    )
+    return data, meta
+
+
+def _get_meteonorm(
+    latitude,
+    longitude,
+    start,
+    end,
+    api_key,
+    parameters,
+    surface_tilt,
+    surface_azimuth,
+    time_step,
+    horizon,
+    interval_index,
+    map_variables,
+    url,
+    endpoint,
+):
```

Comment:
We don't use this formatting anywhere else in pvlib, do we?  I'm not a fan of it personally.